lib/auth.ts

```diff
@@ -27,30 +24,54 @@ export const {
   },
   providers: [
     Credentials({
-      async authorize(credentials, req) {
-        const email = credentials.email as string;
-        const password = credentials.password as string;
-        if (!password) return null;
+        
+      async authorize(credentials) {
+        
+        const { email, password, otp } = credentials;
+        
+        
+        if (!email) return null;
 
-        const users = await getUserFull(email as string);
+        const users = await getUserFull(email);
         if (users.length === 0) return null;
 
         const user = users[0];
-        if (!user.password) return null;
+        
+        // Verify OTP if provided
+        if (otp) {
+
+          const phone = user.phone;
+          
+          if (!phone) return null;
+          const otpIsValid = await verifyOtp(phone, otp);
+      
+          if (otpIsValid) {
+            
+            const authuser: User = {
```

Comment:
can move this below the password check too and do a 
`if (otpIsValid || passwordsMatch) ... return authuser`